nuxeo-distribution/nuxeo-nxr-server/src/main/resources/templates/common-base/client/scripts/upgrade_tomcat7.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+##
+## (C) Copyright 2010-2017 Nuxeo SA (http://nuxeo.com/) and contributors.
+##
+## All rights reserved. This program and the accompanying materials
+## are made available under the terms of the GNU Lesser General Public License
+## (LGPL) version 2.1 which accompanies this distribution, and is available at
+## http://www.gnu.org/licenses/lgpl-2.1.html
+##
+## This library is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## Contributors:
+##     Frantz Fischer
+##
+## Shell script that upgrades Tomcat 7 in Nuxeo (for 6.0, LTS 2015, LTS 2016, LTS 2017)
+## Using explicitely bash to avoid dash which is not POSIX.
+##
+## Requires the following commands to be available: wget, sed, tar, md5sum, shasum
+
+readonly TOMCAT_ARCHIVE_URL="https://archive.apache.org/dist/tomcat/tomcat-7"
+readonly TOMCAT_LATEST_URL="http://www.apache.org/dist/tomcat/tomcat-7"
+
+usage() {
+  echo "Usage:"
+  echo -e "\t./upgrade_tomcat7.sh NUXEO_HOME [TOMCAT_TARGET_VERSION]"
+  echo
+  echo "Note: If no target version is specified the latest one will be retrieved from the Tomcat site"
+  echo
+  echo "Examples:"
+  echo -e "\t./upgrade_tomcat7.sh /Users/ffischer/nuxeo-cap-7.10-tomcat"
+  echo -e "\t./upgrade_tomcat7.sh /Users/ffischer/nuxeo-cap-7.10-tomcat 7.0.76"
+}
+
+if [ $# -eq 0 ] || [ $# -gt 3 ]; then
+  usage
+  exit 1
+fi
+
+# does the Nuxeo location seem valid?
+if [ ! -f "$1/templates/nuxeo.defaults" ]; then
+  echo "ERROR: Cannot find nuxeo.defaults file. Please check the Nuxeo location."
+  echo
+  exit 1
+fi
+NUXEO_HOME=$1
+
+if [ -z "$2" ]; then
+  # autodetects latest version
+  TOMCAT_TARGET=$(curl -sSL ${TOMCAT_LATEST_URL} | grep href=\"v | sed 's/.*href="v\(7.0.[0-9]*\).*/\1/g')
+  TOMCAT_TARGET=${TOMCAT_TARGET:-7.0.75} # fallback to 7.0.75 if autodetection failed
+else
+  # check the TOMCAT version exists
+  VERSIONS_FOUND=$(curl -sSL ${TOMCAT_ARCHIVE_URL} | grep -c "${2}")
+  if [ "$VERSIONS_FOUND" -ne 1 ]; then
+    echo -e "Cannot find Tomcat version ${2}"
+    echo
+    exit 1
+  else
+    TOMCAT_TARGET=$2
+  fi
+fi
+
+if ! WORK_FOLDER=$(mktemp -d -q /tmp/"$(basename "$0")".XXXXXX); then
```

Comment:
The `"$(basename "$0")"` seems overkill, just use `upgrade_tomcat` or something.